### ISSUE TITLE
Licensing of Unicode-related functionality?

### ISSUE THREAD

**Issue by anp** [rust-lang/rust#98116]
In the course of reviewing licenses for Fuchsia's external dependencies, we've noticed that some crates in the ecosystem, [including std](https://github.com/rust-lang/rust/blob/master/src/tools/unicode-table-generator/src/unicode_download.rs), make use of [Unicode data files](https://www.unicode.org/Public/UCD/latest/ucd), but don't seem to treat the code generated from those data files as bound by the [Unicode license](https://www.unicode.org/license.txt).

IANAL, so I'm not sure whether the Unicode license should apply to the [files in core](https://github.com/rust-lang/rust/blob/master/library/core/src/unicode/unicode_data.rs#L1) which seem to be generated from the Unicode data files, but I wanted to raise the possibility for the project to consider.

**Comment by Manishearth** [comment]
Yeah I think it's probably a mistake for unicode-rs to not have the Unicode-DFS license. We inherited the license from back when these crates were a part of rust's stdlib code though.

Unicode-DFS is basically MIT (but applies to data too), so it's not really a problem, but we might need to relicense? There's also iffiness on the licenseability of data though.

**Comment by c-git** [comment]
Have we raised this to the rust foundation for review by a lawyer?

**Comment by Manishearth** [comment]
I believe @pnkfelix has been looking into that

**Comment by c-git** [comment]
I contacted the Rust Foundation and the below is an excerpt from their response:

> Further to your email, I had a chat with our legal counsel just to confirm my own understanding. 
> 
> It’s very common for an open source project (and the corresponding package) to declare a single license for the project, despite including source code for any number of dependencies under other compatible licenses. Take the Linux kernel, for example: it’s primary license is GPLv2, but it contains code licensed under any number of permissive licenses that are compatible with the GPL.
> 
> To comply with these licenses, it’s typically necessary to include a complete copy of each license any time the project is distributed in source or executable form. The Rust project includes the Unicode license in its COPYRIGHT file for this reason: 
> https://github.com/rust-lang/rust/blob/master/COPYRIGHT. Likewise, anyone distributing Rust binaries should include a copy of the Rust COPYRIGHT file so they comply with all of the applicable open source licenses.
> 
> That said, it’s not necessary to declare every license that applies to the codebase in the “license” metadata field for the package. That field should be understood to mean something like: “this is the overall license for the package, but other compatible (i.e. similarly or more permissive) licenses may apply to some portions.” Comprehensive licensing information should be kept in an obvious place in the repository (and corresponding packages), usually the LICENSE or COPYING file. In Rust’s case, it’s COPYRIGHT, LICENSE-APACHE, and LICENSE-MIT.
> 
> In short: it’s not necessary to add the Unicode license to the crate’s license Metadata. The Unicode license should be reproduced in any crate containing Unicode-licensed code.
>
> Hope that answers your question
